### ISSUE TITLE
FileNotFoundException: Too many open files

### ISSUE THREAD

**Issue by thomasbilk** [beryx-gist/badass-jlink-example-kotlin-tornadofx#1]
First off: Thank you for this amazing plugin and this very helpful example repository. 

I just tried this on a Ubuntu 18.04 64 bit Linux with the [AdoptOpenJDK](https://adoptopenjdk.net/) 11 and got this error: 

`java.io.FileNotFoundException: ~/badass-jlink-example-kotlin-tornadofx/build/jlinkbase/mergedjars/kotlin/reflect/jvm/internal/impl/metadata/ProtoBuf$PackageOrBuilder.class (Too many open files)`

Do I need to make an adjustment in my system? What else can I do to run this project successfully?

**Comment by siordache** [comment]
Can you please execute:

```
./gradlew -is jlinkZip
```

and post the output here?

**Comment by thomasbilk** [comment]
Here is the complete output: 

```
Initialized native services in: ~/.gradle/native
Using 4 worker leases.
Starting Build
Settings evaluated using settings file '~/badass-jlink-example-kotlin-tornadofx/settings.gradle'.
Projects loaded. Root project using build file '~/badass-jlink-example-kotlin-tornadofx/build.gradle.kts'.
Included projects: [root project 'badass-jlink-example-kotlin-tornadofx']

> Configure project :
Evaluating root project 'badass-jlink-example-kotlin-tornadofx' using build file '~/badass-jlink-example-kotlin-tornadofx/build.gradle.kts'.
Found module name 'test.kotlin'
------------------------------------------------------------------------
Detecting the operating system and CPU architecture
------------------------------------------------------------------------
os.detected.name=linux
os.detected.arch=x86_64
os.detected.version=4.15
os.detected.version.major=4
os.detected.version.minor=15
os.detected.release=ubuntu
os.detected.release.version=18.04
os.detected.release.like.ubuntu=true
os.detected.release.like.debian=true
os.detected.classifier=linux-x86_64
All projects evaluated.
Selected primary task 'jlinkZip' from project :
Tasks to be executed: [task ':discoverMainScriptsExtensions', task ':compileKotlin', task ':compileJava', task ':processResources', task ':classes', task ':inspectClassesForKotlinIC', task ':jar', task ':prepareMergedJarsDir', task ':createMergedModule', task ':createDelegatingModules', task ':prepareModulesDir', task ':jlink', task ':jlinkZip']
:discoverMainScriptsExtensions (Thread[Execution worker for ':',5,main]) started.

> Task :discoverMainScriptsExtensions
Task ':discoverMainScriptsExtensions' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
:discoverMainScriptsExtensions (Thread[Execution worker for ':',5,main]) completed. Took 0.133 secs.
:compileKotlin (Thread[Execution worker for ':' Thread 3,5,main]) started.

> Task :compileKotlin UP-TO-DATE
Skipping task ':compileKotlin' as it is up-to-date.
:compileKotlin (Thread[Execution worker for ':' Thread 3,5,main]) completed. Took 1.067 secs.
:compileJava (Thread[Execution worker for ':' Thread 3,5,main]) started.

> Task :compileJava
Task ':compileJava' is not up-to-date because:
  Task ':compileJava' has an additional action that was implemented by the Java lambda 'org.javamodularity.moduleplugin.tasks.CompileTask$$Lambda$194/0x0000000800612440'. Using Java lambdas is not supported, use an (anonymous) inner class instead.
file or directory '~/badass-jlink-example-kotlin-tornadofx/src/test/resources', not found
file or directory '~/badass-jlink-example-kotlin-tornadofx/src/test/java', not found
file or directory '~/badass-jlink-example-kotlin-tornadofx/src/test/kotlin', not found
All input files are considered out-of-date for incremental task ':compileJava'.
Full recompilation is required because no incremental change information is available. This is usually caused by clean builds or changing compiler arguments.
Compiling with JDK Java compiler API.
Created classpath snapshot for incremental compilation in 0.177 secs. 1 duplicate classes found in classpath (see all with --debug).
:compileJava (Thread[Execution worker for ':' Thread 3,5,main]) completed. Took 1.471 secs.
:processResources (Thread[Execution worker for ':' Thread 3,5,main]) started.

> Task :processResources NO-SOURCE
file or directory '~/badass-jlink-example-kotlin-tornadofx/src/main/resources', not found
Skipping task ':processResources' as it has no source files and no previous output files.
:processResources (Thread[Execution worker for ':' Thread 3,5,main]) completed. Took 0.011 secs.
:classes (Thread[Execution worker for ':' Thread 3,5,main]) started.

> Task :classes
Skipping task ':classes' as it has no actions.
:classes (Thread[Execution worker for ':' Thread 3,5,main]) completed. Took 0.0 secs.
:inspectClassesForKotlinIC (Thread[Execution worker for ':' Thread 3,5,main]) started.

> Task :inspectClassesForKotlinIC UP-TO-DATE
Skipping task ':inspectClassesForKotlinIC' as it is up-to-date.
:inspectClassesForKotlinIC (Thread[Execution worker for ':' Thread 3,5,main]) completed. Took 0.007 secs.
:jar (Thread[Execution worker for ':' Thread 3,5,main]) started.

> Task :jar
Task ':jar' is not up-to-date because:
  Value of input property 'rootSpec$1.filteringCharset' has changed for task ':jar'
  Value of input property 'rootSpec$2$1.filteringCharset' has changed for task ':jar'
  Value of input property 'rootSpec$2.filteringCharset' has changed for task ':jar'
:jar (Thread[Execution worker for ':' Thread 3,5,main]) completed. Took 0.147 secs.
:prepareMergedJarsDir (Thread[Execution worker for ':' Thread 3,5,main]) started.

> Task :prepareMergedJarsDir UP-TO-DATE
Skipping task ':prepareMergedJarsDir' as it is up-to-date.
:prepareMergedJarsDir (Thread[Execution worker for ':' Thread 3,5,main]) completed. Took 0.743 secs.
:createMergedModule (Thread[Execution worker for ':' Thread 2,5,main]) started.

> Task :createMergedModule
Task ':createMergedModule' is not up-to-date because:
  Task has failed previously.
taskData: org.beryx.jlink.data.CreateMergedModuleTaskData(forceMergedJarPrefixes:[], extraDependenciesPrefixes:[javafx], mergedModuleName:badass.jlink.example.kotlin.tornadofx.merged.module, mergedModuleInfo:, useJdeps:no, mergedJarsDir:~/badass-jlink-example-kotlin-tornadofx/build/jlinkbase/mergedjars, javaHome:~/.sdkman/candidates/java/11.0.1-open, mergedModuleJar:~/badass-jlink-example-kotlin-tornadofx/build/jlinkbase/jlinkjars/badass.jlink.example.kotlin.tornadofx.merged.module.jar, nonModularJarsDirPath:~/badass-jlink-example-kotlin-tornadofx/build/jlinkbase/nonmodjars, jlinkJarsDirPath:~/badass-jlink-example-kotlin-tornadofx/build/jlinkbase/jlinkjars, tmpMergedModuleDirPath:~/badass-jlink-example-kotlin-tornadofx/build/jlinkbase/tmpmerged, tmpModuleInfoDirPath:~/badass-jlink-example-kotlin-tornadofx/build/jlinkbase/tmpmodinfo, tmpJarsDirPath:~/badass-jlink-example-kotlin-tornadofx/build/jlinkbase/tmpjars)
Starting process 'command '~/.sdkman/candidates/java/11.0.1-open/bin/jar''. Working directory: ~/badass-jlink-example-kotlin-tornadofx Command: ~/.sdkman/candidates/java/11.0.1-open/bin/jar --create --file ~/badass-jlink-example-kotlin-tornadofx/build/jlinkbase/tmpmerged/badass.jlink.example.kotlin.tornadofx.merged.module.jar --no-manifest -C ~/badass-jlink-example-kotlin-tornadofx/build/jlinkbase/mergedjars .
Successfully started process 'command '~/.sdkman/candidates/java/11.0.1-open/bin/jar''
Generating module-info in ~/badass-jlink-example-kotlin-tornadofx/build/jlinkbase/tmpjars...
extraDeps: [~/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-base/11.0.1/f1354a284f4151d20358e776f6ff68ee35bbb96d/javafx-base-11.0.1.jar, ~/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-base/11.0.1/4cb7db1adb33b2c0edc458917158664f0107365c/javafx-base-11.0.1-linux.jar, ~/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-controls/11.0.1/61cf91bf3494d0616216f49c9e1d183d170adf0a/javafx-controls-11.0.1.jar, ~/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-controls/11.0.1/e26db61c03faefb09030a5b7b03fa8ce55bb3d04/javafx-controls-11.0.1-linux.jar, ~/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-graphics/11.0.1/e062cb01783effc6413abbd94d1838f6b0add209/javafx-graphics-11.0.1.jar, ~/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-graphics/11.0.1/bcd06886f9fe50da362537eb0e779354e5c38181/javafx-graphics-11.0.1-linux.jar, ~/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-fxml/11.0.1/616cebdad44013674175c710427d0ef94b96bffd/javafx-fxml-11.0.1-linux.jar]
modularJars: [javafx-base-11.0.1.jar, javafx-base-11.0.1-linux.jar, javafx-controls-11.0.1.jar, javafx-controls-11.0.1-linux.jar, javafx-graphics-11.0.1.jar, javafx-graphics-11.0.1-linux.jar, javafx-fxml-11.0.1-linux.jar, javax.json-1.1.2.jar, javax.json-api-1.1.2.jar]
nonModularJars: [kotlin-stdlib-1.3.10.jar, kotlin-stdlib-common-1.3.10.jar, annotations-13.0.jar, kotlin-reflect-1.3.10.jar, tornadofx-1.7.17.jar]
modularJarsRequiredByNonModularJars: [javafx-base-11.0.1.jar, javafx-base-11.0.1-linux.jar, javafx-controls-11.0.1.jar, javafx-controls-11.0.1-linux.jar, javafx-graphics-11.0.1.jar, javafx-graphics-11.0.1-linux.jar, javafx-fxml-11.0.1-linux.jar, javax.json-1.1.2.jar, javax.json-api-1.1.2.jar]
dependenciesHandledAsNonModular: []
Cannot find module exporting javafx.scene.web
Cannot find module exporting org.apache.http.client.methods
Cannot find module exporting org.apache.http.client.protocol
Cannot find module exporting org.apache.http.impl.client
Cannot find module exporting org.osgi.framework
Cannot find module exporting org.osgi.service.url
Cannot find module exporting org.osgi.util.tracker

> Task :createMergedModule FAILED
:createMergedModule (Thread[Execution worker for ':' Thread 2,5,main]) completed. Took 6.207 secs.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.0/userguide/command_line_interface.html#sec:command_line_warnings
7 actionable tasks: 4 executed, 3 up-to-date

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.0/userguide/command_line_interface.html#sec:command_line_warnings
```

**Comment by siordache** [comment]
Thanks! 
I just released the version 2.3.0 of the plugin, which fixes this issue and I configured this example project to use the fixed version.

**Comment by thomasbilk** [comment]
Wow, that was fast. I just tried it and now it builds without any problems. Thank you very much!!
